aiida_vasp/workchains/vasp.py

```diff
@@ -133,12 +139,22 @@ def define(cls, spec):
                        'ERROR_MISSING_PARAMETER_DETECTED',
                        message='the parameter massager did not find expected tags in the input parameters.')
 
+    def _init_parameters(self):
+        """Collect input to the workchain in the converge namespace and put that into the parameters."""
+
+        # At some point we will replace this with possibly input checking using the PortNamespace on
+        # a dict parameter type. As such we remove the workchain input parameters as node entities. Much of
+        # the following is just a workaround until that is in place in AiiDA core.
+        parameters = inherit_and_merge_parameters(self.inputs)
```

Comment:
Originally I did not execute the `inherit_and_merge_parameters` in the VASP workchain as everything should be ready to go when we reach this point. Maybe that is now not the case?

Reply:
I think that when the `dynamics` namespace was added I found that if I did not use the `inherit_and_merge_parameters` not all the options were being passed properly. I can take another look and see if one can remove it

Reply:
yes, please have a look to see if there are some corner cases that should be solved differently

Reply:
One solution is to just set the `self.ctx.inputs.parameters` in the following way, `self.ctx.inputs.parameters = self.inputs.parameters` and  `self.inputs.parameters.dynamics = self.inputs.dynamics` instead of the call to `inherit_and_merge_parameters`

I'm just unsure on whether this would be the best solution

Reply:
Yes, okey, let us leave this. This is needed if we also allow an input on the VASP workchain and do not assume that the `dynamics.selective` is already present in `parameters`. Since we allow that kind of behaviour on our other workchains, we need to follow the same principle. Thanks.